adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/RestAdapterGenerator.java

```diff
@@ -1,66 +1,49 @@
 import uk.gov.justice.raml.core.Generator;
 import uk.gov.justice.raml.core.GeneratorConfig;
-import uk.gov.justice.services.adapters.rest.validator.BaseUriRamlValidator;
-import uk.gov.justice.services.adapters.rest.validator.ResponseContentTypeRamlValidator;
-import uk.gov.justice.services.core.annotation.Component;
+import uk.gov.justice.services.adapters.rest.generator.strategy.AdapterGenerationStrategy;
+import uk.gov.justice.services.adapters.rest.generator.strategy.AdapterGenerationStrategyFactory;
 
 import java.io.File;
 import java.io.IOException;
 import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
+import java.util.LinkedList;
 import java.util.List;
 
 import com.squareup.javapoet.JavaFile;
 import com.squareup.javapoet.TypeSpec;
 import org.raml.model.Raml;
-import org.raml.model.Resource;
 
 public class RestAdapterGenerator implements Generator {
 
-    private static final String PATH_SEPARATOR = "/";
-
-    private final RamlValidator validator = new CompositeRamlValidator(
-            new ContainsResourcesRamlValidator(),
-            new ContainsActionsRamlValidator(),
-            new RequestContentTypeRamlValidator(),
-            new ResponseContentTypeRamlValidator(),
-            new BaseUriRamlValidator()
-    );
-
     @Override
     public void run(final Raml raml, final GeneratorConfig configuration) {
-        validate(configuration);
-        validator.validate(raml);
+        AdapterGenerationStrategy generationStrategy = AdapterGenerationStrategyFactory.createFrom(configuration);
 
-        final Collection<Resource> resources = raml.getResources().values();
-        final Component component = componentFromBaseUriIn(raml);
+        validate(configuration);
+        generationStrategy.validator().validate(raml);
 
         final JaxRsInterfaceGenerator interfaceGenerator = new JaxRsInterfaceGenerator();
         final JaxRsImplementationGenerator implementationGenerator = new JaxRsImplementationGenerator(configuration);
         final JaxRsApplicationCodeGenerator applicationGenerator = new JaxRsApplicationCodeGenerator(configuration);
 
-        writeToResourcePackage(interfaceGenerator.generateFor(resources), configuration);
-        final List<String> implementationNames = writeToResourcePackage(implementationGenerator.generateFor(resources, component), configuration);
+        writeToBasePackage(interfaceGenerator.generateFor(raml), configuration, RESOURCE_PACKAGE_NAME);
+        final List<String> implementationNames = writeToBasePackage(
+                implementationGenerator.generateFor(raml), configuration, RESOURCE_PACKAGE_NAME);
 
-        writeToBasePackage(applicationGenerator.createApplication(implementationNames, raml), configuration);
+        writeToBasePackage(singletonList(applicationGenerator.generateFor(raml, implementationNames)), configuration, "");
```

Comment:
What's wth the empty string? - Is there a cleaner way of doing this? (overloaded method?)